### PR TITLE
Fixes for PN group + dismiss notification once tapped

### PR DESCRIPTION
**Pull request by lachlan-apthletic**
Feature: better grouping of PN notifications
Bug: notifications weren't being dismissed when tapping on them

### CODE REVIEW

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -189,25 +179,94 @@ public void sendToNotificationCentre(Bundle bundle) {
...
+                        extras = notif.getNotification().extras;
+                    }
+                }
+
+                ArrayList<String> existingUsernames = extras.getStringArrayList("existingUsernames");
+                if (existingUsernames == null) {
+                    existingUsernames = new ArrayList<>();
+                }
+                existingUsernames.add(sender);
+                extras.putStringArrayList("existingUsernames", existingUsernames);
```
Comment (ejoncas) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: can you extract "existingUsernames", "existingMessages" and "existingTImestamps" to contstants called `EXTRAS_KEY_USERNAMES`, etc. ?

Comment (lachlan-apthletic) [reply android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: Sure

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 2]
```diff
@@ -43,6 +30,10 @@
     private static final long DEFAULT_VIBRATION = 300L;
     private static final int GROUP_MESSAGE_ID = 0;
```
Comment (ejoncas) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: this is not used?

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 3]
```diff
@@ -43,6 +30,10 @@
     private static final long DEFAULT_VIBRATION = 300L;
```
Comment (ejoncas) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java:30]: not used

**Review by ejoncas — APPROVED** [review]

**Pull request merged**